### PR TITLE
Gfastats

### LINKED ISSUE #2528: new module: GFAStats
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by mahesh-panchal**
Adds the tool GFAstats

## PR checklist

Closes #2528  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by mahesh-panchal** [comment]
@nf-core-bot fix linting

**Review by erinyoung — APPROVED** [review]
This looks great!

It looks like every option of gfastats is covered.

From what I understand, the tests use a SARS-CoV-2 fasta sequence, and then runs gfastats on this fasta using various options, including using a bedfile.

File: modules/nf-core/gfastats/meta.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  - assembly:
+      type: file
+      description: Draft assembly file
+      pattern: "*.{fasta,fastq,gfa}(.gz)?"
+  - out_fmt:
+      type: string
+      description: Output format (fasta, fastq, gfa)
+  - genome_size:
+      type: number
+      description: estimated genome size for NG* statistics (optional).
```
Comment (muffato) [inline modules/nf-core/gfastats/meta.yml:34]:
```suggestion
      description: estimated genome size (bp) for NG* statistics (optional).
```

**Review by muffato — COMMENTED** [review]
tiny winy change

File: modules/nf-core/gfastats/meta.yml (hunk 1 above)
Comment (mahesh-panchal) [reply modules/nf-core/gfastats/meta.yml:34]: Hehe. Good call. Ironically I made the same comment to the GFA author for their repo.

**Pull request merged**